code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/PropositionItem.java

```diff
@@ -0,0 +1,130 @@
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging.internal;
+
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.EventDataKeys.RulesEngine.MESSAGE_CONSEQUENCE_DETAIL_CONTENT;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.EventDataKeys.RulesEngine.MESSAGE_CONSEQUENCE_DETAIL_CONTENT_TYPE;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.EventDataKeys.RulesEngine.MESSAGE_CONSEQUENCE_DETAIL_EXPIRY_DATE;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.EventDataKeys.RulesEngine.MESSAGE_CONSEQUENCE_DETAIL_INBOUND_ITEM_TYPE;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.EventDataKeys.RulesEngine.MESSAGE_CONSEQUENCE_DETAIL_METADATA;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.EventDataKeys.RulesEngine.MESSAGE_CONSEQUENCE_DETAIL_PUBLISHED_DATE;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.EventDataKeys.RulesEngine.MESSAGE_CONSEQUENCE_ID;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.LOG_TAG;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.PayloadKeys.CONTENT;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.PayloadKeys.DATA;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.PayloadKeys.ID;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.PayloadKeys.SCHEMA;
+
+import com.adobe.marketing.mobile.PropositionEventType;
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.util.DataReader;
+import com.adobe.marketing.mobile.util.DataReaderException;
+import com.adobe.marketing.mobile.util.JSONUtils;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.lang.ref.WeakReference;
+import java.util.Map;
+
+/**
+ * A {@link PropositionItem} object contains the experience content delivered within an {@link Inbound} payload.
+ */
+class PropositionItem {
+    private final static String SELF_TAG = "PropositionItem";
+    // Unique proposition identifier
+    private final String uniqueId;
+    // PropositionItem schema string
+    private final String schema;
+    // PropositionItem data content e.g. html or plain-text string or string containing image URL, JSON string
+    private final String content;
+    // Weak reference to Proposition instance
+    WeakReference<Proposition> proposition;
+
+    PropositionItem(final Map<String, Object> item) throws DataReaderException {
```

Comment:
nit: How about we have the constructor accept the properties and instantiate the class as it normally would. And expose a separate method fromEventData (which would internally invoke the constructor) for this.

public static PropositionItem fromEventData(final Map<String, Object> data) {...}

Reply:
i'll update this similarly to Proposition where the constructor will take the individual properties and the property parsing will be the responsibility of the object creating the PropositionItem.

Reply:
Same feedback as for Proposition class.